strawberryfields/apps/sample.py

```diff
@@ -0,0 +1,240 @@
+"""tuple[str]: Available quantum backends for sampling."""
+
+BACKEND_DEFAULTS = {
+    "remote": False,
+    "backend": "gaussian",
+    "threshold": True,
+    "postselect": 0,
+}
+"""Dict[str, Any]: Dictionary to specify default parameters of options in backend sampling for
+:func:`quantum_sampler`.
+"""
+
+
+def _valid_backend(backend: str) -> bool:
+    r"""Performs a validation check that the requested backend can be run based upon the
+    ``QUANTUM_BACKENDS`` tuple.
+
+    Args:
+        backend (str): requested backend for sampling
+
+    Returns:
+        bool: returns only ``True`` if a valid backend is selected
+    """
+    return backend in QUANTUM_BACKENDS
+
+
+# pylint: disable=expression-not-assigned,pointless-statement
+def quantum_sampler(
+    A: np.ndarray,
+    n_mean: float,
+    samples: int = 1,
+    backend_options: Optional[dict] = None,
+) -> list:
+    r"""Generate samples from GBS
+
+    Perform quantum sampling of adjacency matrix :math:`A` using the Gaussian boson sampling
+    algorithm.
+
+    Sampling can be controlled with the optional ``backend_options`` argument, which should be a
+    dict containing any of the following:
+
+    .. glossary::
+
+        key: ``"remote"``, value: *bool*
+            Performs sampling on a remote server if ``True``. Remote sampling is
+            required for sampling on hardware. If not specified, sampling will be performed locally.
+
+        key: ``"backend"``, value: *str*
+            Requested backend for sampling. Available backends are listed in
+            ``QUANTUM_BACKENDS``, these are:
+
+            - ``"gaussian"``: for simulating the output of a GBS device using the
+            :mod:`strawberryfields.backends.gaussianbackend`
+
+        key: ``"threshold"``, value: *bool*
+            Performs sampling using threshold (on/off click) detectors if ``True``
+            or using photon number resolving detectors if ``False``. Defaults to ``True`` for
+            threshold sampling.
+
+        key: ``"postselect"``, value: *int*
+            Causes samples with a photon number or click number less than the
+            specified value to be filtered out. Defaults to ``0`` if unspecified, resulting in no
+            postselection. Note that the number of samples returned in :func:`quantum_sampler` is
+            still equal to the ``samples`` parameter.
+
+    Args:
+        A (array): the (real or complex) :math:`N \times N` adjacency matrix to sample from
+        n_mean (float): mean photon number
+        samples (int): number of samples; defaults to 1
+        backend_options (dict[str, Any]): dictionary specifying options used by backends during
+        sampling; defaults to :const:`BACKEND_DEFAULTS`
+
+    Returns:
+        list[list[int]]: a list of length ``samples`` whose elements are length :math:`N` lists of
+        integers indicating events (e.g., photon numbers or clicks) in each of the :math:`N`
+        modes of the detector
+    """
```

Comment:
Nice! in terms of the high-level 'wrapper' functions, this one has the clearest docstring and short but essential logic.